jf_agent/git/bitbucket_server.py

```diff
@@ -250,38 +252,43 @@ def get_default_branch_commits(
             pull_since = pull_since_date_for_repo(
                 server_git_instance_info, repo['project']['key'], repo['id'], 'commits'
             )
-            try:
 
-                default_branch = (
-                    api_repo.default_branch['displayId'] if api_repo.default_branch else ''
-                )
-                if verbose:
-                    agent_logging.verbose(f"Beginning download of commits for repo {repo['name']}.")
-                commits = api_project.repos[repo['name']].commits(until=default_branch)
-
-                for j, commit in enumerate(
-                    tqdm(commits, desc=f'downloading commits for {repo["name"]}', unit='commits'),
-                    start=1,
-                ):
-                    with agent_logging.log_loop_iters(logger, 'branch commit inside repo', j, 100):
-                        if verbose:
-                            tqdm.write(
-                                f"[{datetime.now().isoformat()}] Getting {commit['id']} ({repo['name']})"
+            # Determine branches to pull commits from for this repo. If no branches are explicitly
+            # provided in a config, only pull from the repo's default branch.
+            repo_branches = [_get_default_branch_name(api_repo)]
+            additional_branches = included_branches.get(api_repo.get()['name'])
+            if additional_branches:
+                repo_branches.extend(additional_branches)
```

Comment:
any reason not to use `get_branches_for_normalized_repo` here?

Reply:
This is because of an annoying heterogeneity between our implementations of BitBucket Server and Github compared to Gitlab and Bitbucket cloud. 

The latter use adapters which make it so that we are already working with our internal, normalized data structures. The former directly interact with the API results so the fields we care about (repo name in this case) are retrieved differently across APIs. Ideally we would move everything to the adapter pattern for consistency.

Reply:
heterogeneity is a great word, that makes sense. I might just add a comment here for future refactorers (refactorinos?) so they don't attempt to simplify and break stuff.

💯 % agreed about cleaning up our adapter pattern usage here.